lib/kafka-consumer.js

```diff
@@ -413,6 +413,52 @@ KafkaConsumer.prototype.unsubscribe = function() {
   return this;
 };
 
+/**
+ * Read a number of messages from a specific topic and partition.
+ *
+ * Can be useful if the consume performance differs per partition. Consuming
+ * per partition could prevent slow performance on one partition from affecting
+ * the consumption of other partitions.
+ *
+ * To select the right partition it is required to set a topic param, because a
+ * consumer can be subscribed to multiple topics.
+ *
+ * @param {string} topic - Name of topic to read
+ * @param {number} partition - Identifier of partition to read
+ * @param {KafkaConsumer~readCallback} cb - Callback to return when work is done.
+ * @param {object} options - Optional settings for consume
+ */
+ KafkaConsumer.prototype.consumeFromToppar = function(topic, partition, cb, options = {}) {
+  if(typeof topic !== 'string') {
+    throw new TypeError('Topic must be a string');
+  }
+  if(typeof partition !== 'number') {
+    throw new TypeError('Partition must be a number');
+  }
+  if(typeof cb !== 'function') {
+    throw new TypeError('Callback must be a function');
+  }
+  if(typeof options !== 'object') {
+    throw new TypeError('Options must be an object');
+  }
+  if(typeof options.numberOfMessages !== 'number' && typeof options.numberOfMessages !== 'undefined') {
```

Comment:
`typeof x !== 'undefined'` can be `x !== undefined` I think?

Reply:
Ah ofcourse.